### PR TITLE
Fix no build option

### LINKED ISSUE #83: Using NoBuild options will stop deployment process
**Describe the bug**
When adding the no-build flag as an option the deployment will not run.

**To Reproduce**
1. Create a prod build by running `ng build --prod`
2. Run `ng deploy --no-build`
3.  You will be provided with a success message but no upload has been done

**Expected behavior**
When using the --no-build error it will need to look at the output path of the target and upload the content of the output path.

### PR DESCRIPTION
**Pull request by erwinverboekend**
fix #83

### CODE REVIEW

**Comment by Jefiozie** [comment]
Hi @erwinverboekend, sorry for my late response but I will review it somewhere in upcoming days. Thanks for the PR.

File: builder/package.json [hunk 1]
```diff
@@ -38,25 +38,23 @@
...
     "ts-jest": "25.2.1",
     "ts-node": "8.6.2",
     "typescript": "3.7.4"
   },
   "dependencies": {
-    "@angular-devkit/architect": "^0.900.0",
+    "@angular-devkit/architect": "^0.900.3",
     "@angular-devkit/core": ">=8.0.0",
-    "@angular-devkit/schematics": ">=8.0.0",
+    "@angular-devkit/schematics": "^9.0.3",
```
Comment (Jefiozie) [inline builder/package.json]: could we revert these?

**Comment by erwinverboekend** [comment]
Hi @Jefiozie, I totally agree with you on this. 

It actually would have been an issue I wanted to address after the no-build issue was fixed :grin: I didn't want to fix it in this PR, because it is unrelated to this issue. It would be a great addition if users can have multiple configurations for different environments and can easily deploy each of them using the deploy command. For the clarification, would you like to see the feature to be able to have multiple deploy configurations added in this PR?

Ah I will take a look at my prettier settings and fix the format, it looks like a misconfiguration.

File: builder/package.json [hunk 2]
```diff
@@ -15,7 +15,7 @@
   "scripts": {
     "build": "npm run copy && npm run build:schema && tsc",
     "build:schema": "json2ts ng-add/schema.json > ng-add/schema.d.ts && json2ts deploy/schema.json > deploy/schema.d.ts",
-    "copy": "cp ../README.md ./ && cp ../LICENSE ./",
+    "copy": "cpy ../README.md ./ && cpy ../LICENSE ./",
```
[the remark this answers is not in the document]
Comment (erwinverboekend) [reply builder/package.json]: I believe `cp` is not available in Windows CMD. Developers who are on Windows could use PowerShell, but using the cpy-cli package makes it easier for Windows developers.

File: builder/package.json (hunk 1 above)
Comment (erwinverboekend) [reply builder/package.json]: Ah didn't want to update the versions here

File: builder/package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (Jefiozie) [reply builder/package.json]: okay great idea, I normally use bash on my windows so never seen this problem.

**Comment by Jefiozie** [comment]
Let's do that change in a separate PR, for now, I think the support of `no build` is the first step.

Let me give you some feedback on your PR specifically for the `no build` feature.

Couldn't we use the code below for getting the target:
```ts
 const target = workspace.getProjectTargets(context.target.project);
const outputPath = target.build.options.outputPath
```
This way we don't have to add the `browserTarget` option to the configuration.

I like all the other changes, looking forward on your changes and merging this PR in!

**Comment by erwinverboekend** [comment]
We could get the `outputPath` in the way you described, but in my opinion there are two reasons why we should not do it this way:

1. I think we should let the deploy builder to be stupid. The deploy builder should not determine which build target the user wants to use by concatenating multiple properties from the configuration. By supplying a target string the deploy builder can find the target and the outputPath that goes along with it in a more standard way without knowing anything about which target actually builds the application.

2. If we use the `browserTarget` option approach, we are prepared for multiple deploy configurations, because users can override their options using the configurations field in their `angular.json`. An example of this approach would be:

```
"deploy": {
    "builder": "@jefiozie/ngx-aws-deploy:deploy",
    "options": {
        "browserTarget": "builder-test:build:production",
        "region": "REGION",
        "bucket": "BUCKET",
        "secretAccessKey": "SECRET",
        "accessKeyId": "ACCESS"
     },
     "configurations": {
         "accept": {
             "browserTarget": "builder-test:build:accept"
         }
     }
}
```

And running the deploy command with the accept configuration:
```
ng deploy -c accept
```

Let me know what you think!

**Comment by Jefiozie** [comment]
Agree on the point where we say the deploy part should be stupid. After some digging in the other deploy builders, I prefer an approach where we have the following setup:

```
"deploy": {
    "builder": "@jefiozie/ngx-aws-deploy:deploy",
	 "configurations": {
        "production": {
            "browserTarget": "project:target:production"
			"region": "YOUR REGION",
			"bucket": "YOUR BUCKET",
        	"secretAccessKey": "YOUR SECRET ACCESSKEY",	            
        	"accessKeyId": "YOUR ACCESS KEY ID"
        }
    }
}
```

Some reasons behind this: 

1. In the case of one app & one configuration, we can use the first one.
2. If we have multiple we can use the `--configuration` flag to specify the specific configuration. This way we can have a specific target for a specific deploy step. And we can (if the user would like to) have different AWS credentials.
3. By default when adding the lib we always should add a `production` configuration to the workspace/project.

**Comment by Jefiozie** [comment]
I'm going to close this one in favor of #173

**Pull request closed without merging**